### PR TITLE
adds debugging for CorruptMessage errors

### PR DESCRIPTION
**Pull request by ronan-mch**
Adds error logging when message corruption errors are received.

### CODE REVIEW

**Comment by dasch** [comment]
I've release `v0.7.1.messagecorruptiondebug` with this code.

**Comment by ronan-mch** [comment]
There's a functional spec failing for some reason, is this normal?

**Comment by dasch** [comment]
Yup

File: lib/kafka/broker.rb [hunk 1]
```diff
@@ -71,7 +71,31 @@ def list_offsets(**options)
...
+      resp.topics.each do |topic_info|
+        errored = topic_info.partitions.find {|part_info| part_info.error_code == 2 }
+        errored_partitions << errored unless errored.nil?
+      end
+
+      errored_partitions.each do |partition_info|
+
+        @logger.error "Corrupt message code received for #{request} to partition #{partition_info.partition}"
+
+        #{"topic" => {"partition" => [message, message] }}
```
Comment (libo) [inline lib/kafka/broker.rb]: do we need this comment?

File: lib/kafka/broker.rb [hunk 2]
```diff
@@ -71,7 +71,31 @@ def list_offsets(**options)
     def produce(**options)
       request = Protocol::ProduceRequest.new(**options)
 
-      send_request(request)
+      resp = send_request(request)
+
+      return if resp.nil?
+
+      errored_partitions = []
```
Comment (libo) [inline lib/kafka/broker.rb]: Would it be possible to gate this logging code behind an `ENV` var that we can set in console? So that we can run it in prod?

**Pull request closed without merging**